src/java.base/share/classes/java/lang/FdLibm.java

```diff
@@ -985,4 +985,213 @@ public static double compute(double x) {
+     *   4. Implementation notes:
+     *      (A). To save one multiplication, we scale the coefficient Qi
+     *           to Qi*2^i, and replace z by (x^2)/2.
+     *      (B). To achieve maximum accuracy, we compute expm1(x) by
+     *        (i)   if x < -56*ln2, return -1.0, (raise inexact if x!=inf)
+     *        (ii)  if k=0, return r-E
+     *        (iii) if k=-1, return 0.5*(r-E)-0.5
+     *        (iv)  if k=1 if r < -0.25, return 2*((r+0.5)- E)
+     *                     else          return  1.0+2.0*(r-E);
+     *        (v)   if (k<-2||k>56) return 2^k(1-(E-r)) - 1 (or exp(x)-1)
+     *        (vi)  if k <= 20, return 2^k((1-2^-k)-(E-r)), else
+     *        (vii) return 2^k(1-((E+2^-k)-r))
+     *
+     * Special cases:
+     *      expm1(INF) is INF, expm1(NaN) is NaN;
+     *      expm1(-INF) is -1, and
+     *      for finite argument, only expm1(0)=0 is exact.
+     *
+     * Accuracy:
+     *      according to an error analysis, the error is always less than
+     *      1 ulp (unit in the last place).
+     *
+     * Misc. info.
+     *      For IEEE double
+     *          if x >  7.09782712893383973096e+02 then expm1(x) overflow
+     *
+     * Constants:
+     * The hexadecimal values are the intended ones for the following
+     * constants. The decimal values may be used, provided that the
+     * compiler will convert from decimal to binary accurately enough
+     * to produce the hexadecimal values shown.
+     */
+    static class Expm1 {
+        private static final double one         =  1.0;
+        private static final double huge        =  1.0e+300;
+        private static final double tiny        =  1.0e-300;
+        private static final double o_threshold =  0x1.62e42fefa39efp9;   //  7.09782712893383973096e+02
+        private static final double ln2_hi      =  0x1.62e42feep-1;       //  6.93147180369123816490e-01
+        private static final double ln2_lo      =  0x1.a39ef35793c76p-33; //  1.90821492927058770002e-10
+        private static final double invln2      =  0x1.71547652b82fep0;   //  1.44269504088896338700e+00
+        /* scaled coefficients related to expm1 */
+        private static final double Q1          = -0x1.11111111110f4p-5;  // -3.33333333333331316428e-02
+        private static final double Q2          =  0x1.a01a019fe5585p-10; //  1.58730158725481460165e-03
+        private static final double Q3          = -0x1.4ce199eaadbb7p-14; // -7.93650757867487942473e-05
+        private static final double Q4          =  0x1.0cfca86e65239p-18; //  4.00821782732936239552e-06
+        private static final double Q5          = -0x1.afdb76e09c32dp-23; // -2.01099218183624371326e-07
+
+        static double compute(double x) {
+            double y, hi, lo, c=0, t, e, hxs, hfx, r1;
+            int k, xsb;
+            /*unsigned*/ int hx;
+
+            hx  = __HI(x);  /* high word of x */
+            xsb = hx & 0x8000_0000;            /* sign bit of x */
+            y = Math.abs(x);
```

Comment:
let's align this variable too
```suggestion
            y   = Math.abs(x);
```